### PR TITLE
Better README

### LINKED ISSUE #589: Update README
Before blog: update verbiage and add pics from blog/paper. 

After blog: add link to blog.

### PR DESCRIPTION
**Pull request by adamjstewart**
This PR updates our README to include similar verbiage and images as our recent [blog](https://pytorch.org/blog/geospatial-deep-learning-with-torchgeo/). At this point, there's a lot of redundancy between the blog and the README. I think that's fine since the examples in the README will continue to be updated long after the blog post is obsolete. But if you feel like it's too similar or have suggestions to make them different let me know.

Closes #589

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+TorchGeo is a [PyTorch](https://pytorch.org/) domain library, similar to [torchvision](https://pytorch.org/vision), providing datasets, samplers, transforms, and pre-trained models specific to geospatial data.
 
 The goal of this library is to make it simple:
 
-1. for machine learning experts to use geospatial data in their workflows, and
-2. for remote sensing experts to use their data in machine learning workflows.
+1. for machine learning experts to work with geospatial data, and
+2. for remote sensing experts to explore machine learning solutions.
 
-See our [installation instructions](#installation), [documentation](#documentation), and [examples](#example-usage) to learn how to use TorchGeo.
```
Comment (adamjstewart) [inline README.md:10]: These links worked on GitHub but not on PyPI. We _could_ fix them, but I'm not sure how useful they actually are since these 3 sections immediately follow the links.

File: README.md [hunk 2]
```diff
@@ -28,113 +26,138 @@ The recommended way to install TorchGeo is with [pip](https://pip.pypa.io/):
...
 ```
 
-### Benchmark datasets
+### Geospatial datasets and samplers
+
+Many remote sensing applications involve working with [*geospatial datasets*](https://torchgeo.readthedocs.io/en/stable/api/datasets.html#geospatial-datasets)—datasets with geographic metadata. These datasets can be challenging to work with due to the sheer variety of data. Geospatial imagery is often multispectral with a different number of spectral bands and spatial resolution for every satellite. In addition, each file may be in a different coordinate reference system (CRS), requiring the data to be reprojected into a matching CRS.
 
-TorchGeo includes a number of [*benchmark*](https://torchgeo.readthedocs.io/en/latest/api/datasets.html#non-geospatial-datasets) datasets, datasets that include both input images and target labels. This includes datasets for tasks like image classification, regression, semantic segmentation, object detection, instance segmentation, change detection, and more.
+<img src="https://raw.githubusercontent.com/microsoft/torchgeo/main/images/geodataset.png" alt="Example application in which we combine Landsat and CDL and sample from both"/>
```
Comment (adamjstewart) [inline README.md:54]: By using absolute links to the files in GitHub, we don't have to include the images in the sdist we upload to PyPI.

File: README.md [hunk 3]
```diff
@@ -28,113 +26,138 @@ The recommended way to install TorchGeo is with [pip](https://pip.pypa.io/):
...
+trainer = Trainer(gpus=1, default_root_dir="...")
+
+trainer.fit(model=task, datamodule=datamodule)
+```
+
+<img src="https://raw.githubusercontent.com/microsoft/torchgeo/main/images/inria.png" alt="Building segmentations produced by a U-Net model trained on the Inria Aerial Image Labeling dataset"/>
 
-We provide a script, `train.py` for training models using a subset of the datasets. We do this with the PyTorch Lightning `LightningModule`s and `LightningDataModule`s implemented under the `torchgeo.trainers` namespace.
-The `train.py` script is configurable via the command line and/or via YAML configuration files. See the [conf/](conf/) directory for example configuration files that can be customized for different training runs.
+In our GitHub repo, we provide `train.py` and `evaluate.py` scripts to train and evaluate the performance of models using these datamodules and trainers. These scripts are configurable via the command line and/or via YAML configuration files. See the [conf](https://github.com/microsoft/torchgeo/blob/main/conf) directory for example configuration files that can be customized for different training runs.
```
Comment (adamjstewart) [inline README.md:139]: Can add `predict.py` to this list once #560 is complete.

File: README.md [hunk 4]
```diff
@@ -28,113 +26,138 @@ The recommended way to install TorchGeo is with [pip](https://pip.pypa.io/):
...
+    title = {{TorchGeo}: Deep Learning With Geospatial Data},
     url = {https://github.com/microsoft/torchgeo},
     year = {2021}
 }
 ```
 
 ## Contributing
 
-This project welcomes contributions and suggestions. If you would like to submit a pull request, see our [Contribution Guide](https://torchgeo.readthedocs.io/en/latest/user/contributing.html) for more information.
+This project welcomes contributions and suggestions. If you would like to submit a pull request, see our [Contribution Guide](https://torchgeo.readthedocs.io/en/stable/user/contributing.html) for more information.
```
Comment (adamjstewart) [inline README.md:161]: Changed all of the documentation links from "latest" to "stable" to match our new default documentation version.

**Review by calebrob6 — APPROVED** [review]
